coderd/users.go

```diff
@@ -569,6 +569,72 @@ func (api *API) userByName(rw http.ResponseWriter, r *http.Request) {
 	httpapi.Write(ctx, rw, http.StatusOK, db2sdk.User(user, organizationIDs))
 }
 
+// Returns recent build parameters for the signed-in user.
+//
+// @Summary Get autofill build parameters for user
+// @ID get-autofill-build-parameters-for-user
+// @Security CoderSessionToken
+// @Produce json
+// @Tags Users
+// @Param user path string true "User ID, username, or me"
+// @Param template_id query string true "Template ID"
+// @Success 200 {array} codersdk.UserParameter
+// @Router /users/{user}/autofill-parameters [get]
+func (api *API) userAutofillParameters(rw http.ResponseWriter, r *http.Request) {
+	var (
+		apiKey = httpmw.APIKey(r)
+		user   = httpmw.UserParam(r)
+	)
+
+	if apiKey.UserID != user.ID {
+		httpapi.Write(r.Context(), rw, http.StatusForbidden, codersdk.Response{
+			Message: "You are not authorized to view this user's parameters.",
+		})
+		return
+	}
```

Comment:
Putting Authz checks in individual HTTP handlers is a code smell.  Authz checks against data access should all happen in the dbauthz layer.

In this case, this check breaks a site admin reading parameters.